core/src/main/java/kafka/server/share/ShareFetchUtils.java

```diff
@@ -161,6 +161,20 @@ static long offsetForLatestTimestamp(TopicIdPartition topicIdPartition, ReplicaM
         return timestampAndOffset.get().offset;
     }
 
+    /**
+     * The method is used to get the offset for the given timestamp for the topic-partition.
+     *
+     * @return The offset for the given timestamp.
+     */
+    static long offsetForTimestamp(TopicIdPartition topicIdPartition, ReplicaManager replicaManager, long timestampToSearch, int leaderEpoch) {
+        Option<FileRecords.TimestampAndOffset> timestampAndOffset = replicaManager.fetchOffsetForTimestamp(
+            topicIdPartition.topicPartition(), timestampToSearch, Option.empty(), Optional.of(leaderEpoch), true).timestampAndOffsetOpt();
+        if (timestampAndOffset.isEmpty()) {
+            throw new OffsetNotAvailableException("offset for timestamp to search: " + timestampToSearch + " not found for topic partition: " + topicIdPartition);
```

Comment:
For consistency, I would adjust the capitalisation. I suggest `"Offset for timestamp " + timestamptoSearch + " not found for topic partition"` instead.

Reply:
Looking at the latest changes, this was not done as requested. Please can you change the wording.